x-pack/test/api_integration/apis/ml/jobs/datafeed_preview.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const jobId = `fq_datafeed_preview_${Date.now()}`;
+
+  const job = {
+    job_id: `${jobId}_1`,
+    description: '',
+    groups: ['automated', 'farequote'],
+    analysis_config: {
+      bucket_span: '30m',
+      detectors: [{ function: 'distinct_count', field_name: 'airline' }],
+      influencers: [],
+    },
+    data_description: { time_field: '@timestamp' },
+    analysis_limits: { model_memory_limit: '11MB' },
+  };
+
+  function isUpperCase(str: string) {
+    return /^[A-Z]+$/.test(str);
+  }
+
+  function isLowerCase(str: string) {
+    return /^[a-z]+$/.test(str);
+  }
```

Comment:
Ff a string contains a number or some special charts it will return false. Have you considered something like example below?
```suggestion
  function isLowerCase(str: string) {
    return !/[A-Z]+/.test(str)
  }
```